CIScripts/Test/TestConfigurationUtils.ps1

```diff
@@ -297,22 +297,27 @@ function Remove-AllUnusedDockerNetworks {
     }
 }
 
-[ScriptBlock]$SelectValidNetIPInterfaceBody = {
-    $_ `
+function Select-ValidNetIPInterface {
+     Param ([parameter(Mandatory=$true, ValueFromPipeline=$true)]$GetIPAddressOutput)
+    
+    Process { $_ `
         | Where-Object AddressFamily -eq "IPv4" `
-        | Where-Object { ($_.SuffixOrigin -eq "Dhcp") -or ($_.SuffixOrigin -eq "Manual") } 
+        | Where-Object { ($_.SuffixOrigin -eq "Dhcp") -or ($_.SuffixOrigin -eq "Manual") }
+    }
 }
 
 function Wait-RemoteInterfaceIP {
     Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session,
            [Parameter(Mandatory = $true)] [String] $AdapterName)
+    $InjectedFunction = [PSCustomObject]@{ Name = 'Select-ValidNetIPInterface'; Body = ${Function:Select-ValidNetIPInterface} }
 
-    Invoke-UntilSucceeds -Name "Waiting for IP on interface $AdapterName" -Duration 60 {
+    Invoke-UntilSucceeds -Name "Waiting for IP on interface $AdapterName" -Duration 15 {
         Invoke-Command -Session $Session {
-            $ScriptBlock = [ScriptBlock]::Create($Using:SelectValidNetIPInterfaceBody)
+            $Using:InjectedFunction | ForEach-Object { Invoke-Expression "function $($_.Name) { $($_.Body) }" }
```

Comment:
[possible future refinement] Wrap this in some kind of `Invoke-CommandWithCaptureFunctions` helper with proper tests etc.

[nit] When using `$( expr )` expression substitution in strings, it's nice to put spaces inside `()`, ie. `$( $_.Name )`. (Not sure where I've read that style rule, but I believe it's not something that I've just invented)